tools/sync/commands.py

```diff
@@ -67,35 +70,68 @@ def __init__(self, base):
         self.logger = logging.getLogger(__name__)
         logging.basicConfig()
 
+    def run_command(command):
+        cmd = Command(command,
+                      args_subst={"PROJECT": self.name},
+                      args_append=[self.name], excl_subst=True)
+        cmd.execute()
+        self.retcodes[str(cmd)] = cmd.getretcode()
+        self.outputs[str(cmd)] = cmd.getoutput()
+
     def run(self):
         """
         Run the sequence of commands and capture their output and return code.
         First command that returns code other than 0 terminates the sequence.
         If the command has return code 2, the sequence will be terminated
         however it will not be treated as error.
+
+        Any command entry that is a URI, will be used to submit RESTful API
+        request. Return codes for these requests are not checked.
         """
 
+        PROJECT_SUBST = '%PROJECT%'
         for command in self.commands:
-            cmd = Command(command,
-                          args_subst={"ARG": self.name},
-                          args_append=[self.name], excl_subst=True)
-            cmd.execute()
-            self.retcodes[str(cmd)] = cmd.getretcode()
-            self.outputs[str(cmd)] = cmd.getoutput()
-
-            # If a command fails, terminate the sequence of commands.
-            retcode = cmd.getretcode()
-            if retcode != 0:
-                if retcode == 2:
-                    self.logger.info("command '{}' requested break".
-                                     format(cmd))
-                    self.run_cleanup()
+            if is_web_uri(command[0]):
+                uri = command[0].replace(PROJECT_SUBST, self.name)
```

Comment:
maybe python builtin format method would be better

`'Coordinates: {latitude}, {longitude}'.format(latitude='37.24N', longitude='-115.81W')`

Reply:
Aha. Nice